### PR TITLE
[vcpkg-cmake] Fix docs and usage of option MAYBE_UNUSED_VARIABLES

### PR DESCRIPTION
**Pull request by JackBoosY**
In the current document, our description of `MAYBE_UNUSED_VARIABLES` caused the user to misunderstand the parameter and incorporate the wrong changes.

Fix the option description and add sample code to the docs.

Related: https://github.com/microsoft/vcpkg/pull/13467

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @wrobelda

File: ports/kf5sonnet/portfile.cmake [hunk 1]
```diff
@@ -14,13 +14,6 @@ vcpkg_cmake_configure(
         -DKDE_INSTALL_PLUGINDIR=plugins
         -DKDE_INSTALL_DATAROOTDIR=data
         -DKDE_INSTALL_QTPLUGINDIR=plugins
-        -DBUILD_HTML_DOCS=OFF
-        -DBUILD_MAN_DOCS=OFF
-        -DBUILD_QTHELP_DOCS=OFF
-    MAYBE_UNUSED_VARIABLES
-        BUILD_HTML_DOCS
-        BUILD_MAN_DOCS
-        BUILD_QTHELP_DOCS
```
Comment (JackBoosY) [inline ports/kf5sonnet/portfile.cmake:23]: @wrobelda I checked the upstream CMakeLists.txt, but these options doesn't exist.

**Comment by JackBoosY** [comment]
Need to check the related warning message after pipeline test.

File: ports/kf5sonnet/portfile.cmake (hunk 1 above)
Comment (wrobelda) [reply ports/kf5sonnet/portfile.cmake:23]: Indeed. However, these are standard KDE variables used across KDE software. If at any point a library adds a manual or HTML docs, these will apply. I thought that it makes sense for consistency with other frameworks where such variables are already present in their CMakeLists.txt.

Comment (JackBoosY) [reply ports/kf5sonnet/portfile.cmake:23]: @wrobelda 
`MAYBE_UNUSED_VARIABLES` is only for the corresponding ports. If some of them have these parameters, we should only add `MAYBE_UNUSED_VARIABLES` to these ports.

**Comment by JackBoosY** [comment]
Question: should we also add the following requirement in https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#build-techniques?

**Review by strega-nil-ms — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**